app/scripts/lib/height-observer.js

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Observe the height of an element, trigger the `change` notification
+ * if the element's height changes.
+ */
+
+'use strict';
+
+define([
+  'backbone',
+  'underscore'
+], function (Backbone, _) {
+  function HeightObserver (options) {
+    options = options || {};
+
+    this._targetEl = options.target;
+    this._window = options.window;
+
+    if ('delayMS' in options) {
+      this._delayMS = options.delayMS;
+    }
+  }
+
+  _.extend(HeightObserver.prototype, Backbone.Events, {
+    _delayMS: 50,
+
+    start: function () {
+      var self = this;
+
+      if (self._observer) {
+        throw new Error('already started');
+      }
+
+      // For more info, see
+      // https://developer.mozilla.org/docs/Web/API/MutationObserver
+      var MutationObserver = self._window.MutationObserver;
+      if (MutationObserver) {
+        var onMutation = _.debounce(self._onMutation.bind(self), self._delayMS);
+        self._observer = new MutationObserver(onMutation);
+
+        self._observer.observe(self._targetEl, {
+          attributes: true,
+          attributeFilter: ['class', 'style'],
+          characterData: true,
+          childList: true,
+          subtree: true
+        });
+      }
+    },
+
+    _lastHeight: -Infinity,
+    _onMutation: function () {
+      var self = this;
+      var currentHeight = self._targetEl.clientHeight;
```

Comment:
Logged https://github.com/mozilla/fxa-content-server/issues/2614 about the undefined/null issue in the metrics.